File: app/src/main/java/github/yaa110/memento/widget/template/ModelViewHolder.java
package github.yaa110.memento.widget.template;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import github.yaa110.memento.R;
import github.yaa110.memento.model.DatabaseModel;

abstract public class ModelViewHolder<T extends DatabaseModel> extends RecyclerView.ViewHolder {
    public View holder;
    public View selected;

    public ModelViewHolder(View itemView) {
        super(itemView);
        holder = itemView.findViewById(R.id.holder);
        selected = itemView.findViewById(R.id.selected);
    }

    public void setSelected(boolean status) {
        selected.setVisibility(status ? View.VISIBLE : View.GONE);
    }

    abstract public void populate(T item);
}

File: app/src/main/java/github/yaa110/memento/widget/template/ModelViewHolder.kt
package github.yaa110.memento.widget.template

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import github.yaa110.memento.R
import github.yaa110.memento.model.DatabaseModel

abstract class ModelViewHolder<T : DatabaseModel?>(itemView: View) :
    RecyclerView.ViewHolder(itemView) {
    var holder: View = itemView.findViewById(R.id.holder)
    var selected: View = itemView.findViewById(R.id.selected)

    fun setSelected(status: Boolean) {
        selected.visibility = if (status) View.VISIBLE else View.GONE
    }

    abstract fun populate(item: T)
}
